docs/resources/function.md

```diff
@@ -0,0 +1,112 @@
+---
+subcategory: "Unity Catalog"
+---
+# databricks_function Resource
+
+-> This resource source can only be used with a workspace-level provider.
+
+Creates a [User-Defined Function (UDF)](https://docs.databricks.com/en/udf/unity-catalog.html) in Unity Catalog. UDFs can be defined using SQL, or external languages (e.g., Python) and are stored within [Unity Catalog schemas](../resources/schema.md). 
+
+## Example Usage
+
+### SQL-based function:
+
+```hcl
+resource "databricks_catalog" "sandbox" {
+  name    = "sandbox_example"
+  comment = "Catalog managed by Terraform"
+}
+
+resource "databricks_schema" "functions" {
+  catalog_name = databricks_catalog.sandbox.name
+  name         = "functions_example"
+  comment      = "Schema managed by Terraform"
+}
+
+resource "databricks_function" "calculate_bmi" {
+  name              = "calculate_bmi"
+  catalog_name      = databricks_catalog.sandbox.name
+  schema_name       = databricks_schema.functions.name
+  input_params      = [
```

Comment:
It could be easier for people to specify `input_param` as separate blocks, like we do in other resources

Reply:
Although doc says that `input_params` is a block, and then there are parameters inside: https://docs.databricks.com/api/workspace/functions/create#input_params